### ISSUE TITLE
Problem with UIPageviewController with moveTo

### ISSUE THREAD

**Issue by IronMaltes** [richardtop/CalendarKit#318]
### New Issue Checklist

- [x] Use the latest version of CalendarKit
- [x] I've read the [Contribution Guidelines](https://github.com/richardtop/CalendarKit/blob/master/CONTRIBUTING.md)
- [x] I've searched for [existing GitHub issues](https://github.com/richardtop/CalendarKit/issues)

### Issue Description

Hi Richard, 

I have a problem once I changed to iOS15, I waited till the iOS15 RC to write this issue.
The problem is that changing days using 'moveTo' or using the header, at the end everything gets messy, half pages, not doing apparently the layout or some other things you can see on the video attached.

I tried removing all my code that modifies the creation of the Event with no success.
I've seen that inside the func move in TimelinePagerView that (pagingViewController.setViewControllers) you implemented some kind of workaround, but seems not to work right now. I did also some research in stackoverflow with little success.

Do you have any advice on this?

Thanks a lot.
Alex.

https://user-images.githubusercontent.com/32038447/133603473-7fc05a0f-2c9d-4612-bbce-81d8eadd9dba.mov

**Comment by IronMaltes** [comment]
After some more research, I found a workaround modifying the calendar kit  function move - pagingViewController.setViewControllers with animate + false (with no hidden reload), just for the moveTo. 

I suspect is related to the time my loadEvents delegate took (now I pre cache the info just for the move to) and also to several animation taking place at the same time. That I think is the main root cause.

What’s your thoughts on that, make sense?

**Comment by richardtop** [comment]
Hi, thanks for the bug report.
Could you please confirm:
1. whether this is reproducible in the demo project in the CalendarKit
2. In the CalendarApp https://github.com/richardtop/CalendarApp
3. please, share the code you're using here.

Once I get this info, I'll be able to assist you.

**Comment by IronMaltes** [comment]
My problem is solved with the workaround I described above. You can close this issue if you want. Nevertheless if you want more info on what a I’ve done I can extend the explanation. 

Is the same problem I had in other parts of my app when using page view control. And also looks the same as many other problems I saw in stack overflow and here, the animations of pageview and other collide (or some processes inside the layout…) or don’t have time to execute properly, but just for the function (setViewController or for instance in calendarkit, moveTo) not for the swipe left or right.

Thanks for your nice work with this pod 😉

**Comment by richardtop** [comment]
Yeah, please share the workaround and the places where you had to put it. Also, links to the SO with the same questions would be helpful.
